services/wallet/following/manager.go

```diff
@@ -0,0 +1,96 @@
+package following
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/services/wallet/thirdparty/efp"
+)
+
+// Manager handles following address operations using EFP providers
+type Manager struct {
+	providers []efp.FollowingDataProvider
+}
+
+// NewManager creates a new following manager with the provided EFP providers
+func NewManager(providers []efp.FollowingDataProvider) *Manager {
+	return &Manager{
+		providers: providers,
+	}
+}
+
+// FetchFollowingAddresses fetches the list of addresses that the given user is following
+// Uses the first available provider (can be enhanced later with fallback logic)
+func (m *Manager) FetchFollowingAddresses(ctx context.Context, userAddress common.Address, search string, limit, offset int) ([]efp.FollowingAddress, error) {
+	logutils.ZapLogger().Debug("following.Manager.FetchFollowingAddresses",
+		zap.String("userAddress", userAddress.Hex()),
+		zap.String("search", search),
+		zap.Int("limit", limit),
+		zap.Int("offset", offset),
+		zap.Int("providers.len", len(m.providers)),
+	)
+
+	if len(m.providers) == 0 {
+		return []efp.FollowingAddress{}, nil
```

Comment:
feels like we should return an error if the provider is not set and we make a request, otherwise we can't distinguish this from a "true" empty FollowingAddresses result.